interop/authzen-todo-backend/src/statefulAuthorizationService.ts

```diff
@@ -0,0 +1,94 @@
+export function createStatefulAuthorizationService( pdpBaseName:string, pdpAuthHeader:string, pdpHeader:string): StatefulAuthorizationService {
+  console.log(pdpHeader);
+  switch (pdpHeader.toLowerCase()) {
+    case "openfga":
+      return new OpenFGAStatefulAuthorizationService(pdpBaseName, pdpAuthHeader);
+    default:
+      return new NullStatefulAuthorizationService();
+  }
+}
+
+export interface StatefulAuthorizationService {
+  insert(todoId: string, userId: string): Promise<void>;
+  delete(todoId: string, userId: string): Promise<void>;
+}
+
+class NullStatefulAuthorizationService implements StatefulAuthorizationService {
+  insert(_todoId: string, _userId: string): Promise<void> {
+    return Promise.resolve();
+  }
+
+  delete(_todoId: string, _userId: string): Promise<void> {
+    return Promise.resolve();
+  }
+}
+
+class OpenFGAStatefulAuthorizationService implements StatefulAuthorizationService {
```

Comment:
I like this interface-based approach - seems nice and clean. 

It may feel a bit like overkill right now since there's only one stateful implementation, but seems like we want a subdirectory with stateful providers, and have each implementation in its own source file?  `statefulAuthorizationService.ts` can import each of them and the constructor can return the imported implementation.

Reply:
You mean keeping

`export function createStatefulAuthorizationService( pdpBaseName:string, pdpAuthHeader:string, pdpHeader:string): StatefulAuthorizationService {
  console.log(pdpHeader);
  switch (pdpHeader.toLowerCase()) {
    case "openfga":
      return new OpenFGAStatefulAuthorizationService(pdpBaseName, pdpAuthHeader);
    default:
      return new NullStatefulAuthorizationService();
  }
}`

but having the 'OpenFGAStatefulAuthorizationService' class in a 'stateful/openfga.ts' file?

Reply:
yes - exactly

Reply:
@ogazitt done